service/controller/cluster.go

```diff
@@ -141,7 +156,14 @@ func newClusterResources(config ClusterConfig) ([]resource.Interface, error) {
 	}
 
 	resources := []resource.Interface{
+		// appFinalizerResource is executed first and removes finalizers after
+		// the per cluster app-operator instance has been deleted.
+		appFinalizerResource,
```

Comment:
There will only be 4 resources once app is added but the order matters.

So I added comments. Feedback welcome if we can improve them.